base/strings/util.jl

```diff
@@ -464,6 +464,55 @@ function hex2bytes(s::AbstractString)
 end
 
 """
+    hex2bytes(d::Vector{UInt8}, s::Vector{UInt8}, nInBytes::Int=length(s))
+
+Convert the first `nInBytes` hexadecimal bytes array to its binary representation. The
+results are populated into a destination array. The function returns the number of bytes
+copied into the destination array. The size of destination array must be at least half of
+the `nInBytes` parameter.
+"""
+function hex2bytes(d::Vector{UInt8}, s::Vector{UInt8}, nInBytes::Int=length(s))
```

Comment:
Since this modifies its argument, it should be called `hex2bytes!`. It would be good to also add a 1-argument version that allocates the output for you, `hex2bytes(s::Vector{UInt8})`.